PowerDNS/pdns
# Spell check 0.0.17a

### Short description

Upgrade the spell checking action

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### Plurals and paste tenses
In the past, plurals `foo`+`s` and past tenses `foo`+`ed` were automatically tolerated. This turned out to be a bad design choice on my part.

The basic example is that `potatos` would sometimes be treated as a mistake and sometimes not (depending on the presence of `potato`).

This is also why `transfered`/`crashs` were accepted as a word in the past, so `transfer` + `ed` / `crash` + `s` were acceptable).

The downside is that plurals of terms of art have to be added individually. Given that the bot helps out w/ this (and the next release should let you have the bot do it for you), this hasn't turned out to be a huge cost.

### Pull requests

When GitHub initially introduced GitHub Actions, the event for `pull_request` was created without enough permission for a tool like this to work properly. In 2020, they introduced a replacement event `pull_request_target` which has enough permission (and thus works better).

### Validation Steps Performed
I've created a temporary repo which is using this as its default branch.

There's a sample PR here:
https://github.com/check-spelling/pdns/pull/1

If people want, they're welcome to create PRs into that repo to test (you can also test in your own forks).
